File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        p_map = { '}': '{', ']':'[', ')':'(' }
        for char in s:
            if char in p_map:
                if stack and stack[-1] == p_map[char]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(char)
        return True if not stack else False

File: Data Structures & Algorithms/daily-temperatures/submission-2.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        
        const res = Array.from({length: temperatures.length}, () => 0);
        const stack = []; // pair: [temp, index]

        for (let i = 0; i < temperatures.length; i++){
            const t = temperatures[i];
            while(stack.length > 0 && t > stack[stack.length-1][0]){
                const [_, idx] = stack.pop();
                res[idx] = i - idx;
            }
            stack.push([t, i]);
        }
        return res;
    }
}
